File: Problems/0217.Contains-Duplicate/ContainsDuplicate.py
from typing import List

class Solution:

    # Runtime: 120 ms, faster than 95.29% of Python3 online submissions
    def containsDuplicate(self, nums: List[int]) -> bool:
        rec = set()
        for n in nums:
            if n in rec:
                return True
            rec.add(n)
        return False

    # Runtime: 120 ms, faster than 95.29% of Python3 online submissions
    def containsDuplicate_v2(self, nums: List[int]) -> bool:
        return len(set(nums)) != len(nums)

File: leetcode/0217.Contains-Duplicate/ContainsDuplicate.go
package leetcode

// Runtime: 12 ms, faster than 100.00% of Go online submissions
func containsDuplicate(nums []int) bool {
	rec := make(map[int]bool, len(nums))
	for _, n := range nums {
		if _, isExist := rec[n]; isExist {
			return true
		}
		rec[n] = true
	}
	return false
}
